src/indexing/enclave_lfs_access.h

```diff
@@ -0,0 +1,282 @@
+            if (result != nullptr)
+            {
+              if (result->fetch_result == FetchResult::Fetching)
+              {
+                const auto success = verify_and_decrypt(
+                  *encryption_key,
+                  obfuscated,
+                  std::move(encrypted),
+                  result->contents);
+                if (success)
+                {
+                  result->fetch_result = FetchResult::Loaded;
+                }
+                else
+                {
+                  result->fetch_result = FetchResult::Corrupt;
+                  LOG_TRACE_FMT(
+                    "Cache was given invalid contents for {} (aka {})",
+                    obfuscated,
+                    result->key);
+                }
+              }
+              else
+              {
+                LOG_FAIL_FMT(
+                  "Retained result for {} (aka {}) in state {}",
+                  obfuscated,
+                  result->key,
+                  result->fetch_result);
+              }
+            }
+            else
+            {
+              LOG_TRACE_FMT(
+                "Received response for {}, but caller has already dropped "
+                "result",
+                obfuscated);
+            }
+            pending.erase(it);
+          }
+          else
+          {
+            LOG_TRACE_FMT(
+              "Ignoring response message for unrequested key {}", obfuscated);
+          }
+        });
+
+      DISPATCHER_SET_MESSAGE_HANDLER(
+        dispatcher,
+        LFSMsg::not_found,
+        [this](const uint8_t* data, size_t size) {
+          auto [obfuscated] =
+            ringbuffer::read_message<LFSMsg::not_found>(data, size);
+          auto it = pending.find(obfuscated);
+          if (it != pending.end())
+          {
+            auto result = it->second.lock();
+            if (result != nullptr)
+            {
+              if (result->fetch_result == FetchResult::Fetching)
+              {
+                LOG_TRACE_FMT(
+                  "Host has no contents for key {} (aka {})",
+                  obfuscated,
+                  result->key);
+                result->fetch_result = FetchResult::NotFound;
+              }
+              else
+              {
+                LOG_FAIL_FMT(
+                  "Retained result for {} (aka {}) in state {}",
+                  obfuscated,
+                  result->key,
+                  result->fetch_result);
+              }
+            }
+            else
+            {
+              LOG_TRACE_FMT(
+                "Received not_found for {}, but caller has already dropped "
+                "result",
+                obfuscated);
+            }
+            pending.erase(it);
+          }
+          else
+          {
+            LOG_TRACE_FMT(
+              "Ignoring not_found message for unrequested key {}", obfuscated);
+          }
+        });
+    }
+
+    static inline LFSKey obfuscate_key(const LFSKey& key)
```

Comment:
I realised that as well as encryption, we maybe need some kind of key obfuscation? We should discuss exactly what kind of information leakage to the host with these index files is acceptable/inevitable (additional things we could add - a salt to this hash, and padding to the encryption).

Reply:
In terms of naming, I think this is more than just obfuscation as we hash the `key` here?

Reply:
Let's keep this public-only until we've reviewed the full implications on confidentiality, just digesting the key + range sounds vulnerable to dictionary attacks.